artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3159,6 +3162,9 @@ public void run() {
                         connection.killMessage(server.getNodeID());
                         remotingService.removeConnection(connection.getID());
                         connection.fail(ActiveMQMessageBundle.BUNDLE.connectionsClosedByManagement(connection.getRemoteAddress()));
+                        //break once a consumer gets killed. This can prevent all
+                        //consumers to this queue get killed all at once.
+                        break;
```

Comment:
this fix is not valid....
We need all the slow consumers to be killed. 
We need to calculate the rate properly then. threshold / number of consuemrs for instance.